ukaea/piezo
# Feature/add optional label input/#52

Adding optional argument when submitting a job of `label`. This label can be used to filter the sparkapplications on kubernetes. The label has the key `userLabel` and is only present if specified. All system tests run with the label `userLabel=systemTest`

**Note this branch was built off #45 which should be merged in before this one**

### Use case:
Cleaning up system tests after running: `kubectl delete sparkapplications -l userLabel=systemTest`
Potentially used as a unique identifier in the future where could be used as the user name .

### To test:
* Run unit and system tests.
* Run a seperate job through postman  without the systemtest label
* `kubectl delete sparkapplications -l userLabel=systemTest` check only system test jobs are deleted and not the one run through postman


### Stories covered
#52
